themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/signingin-page/SigningInPage.tsx

```diff
@@ -68,116 +92,202 @@ class SigningInPage extends React.Component<SigningInPageProps, SigningInPageSta
     public constructor(props: SigningInPageProps) {
         super(props);
         this.state = {
-            twoFactorEnabled: true,
-            twoFactorEnabledText: Msg.localize('twoFactorEnabled'),
-            isTotpConfigured: features.isTotpConfigured,
+            credentialContainers: new Map()
         }
-        this.setLastPwdUpdate();
+
+        this.getCredentialContainers();
     }
 
-    private setLastPwdUpdate(): void {
-        AccountServiceClient.Instance.doGet("/credentials/password")
-            .then((response: AxiosResponse<PasswordDetails>) => {
-                if (response.data.lastUpdate) {
-                    const lastUpdate: number = response.data.lastUpdate;
-                    this.setState({lastPasswordUpdate: lastUpdate});
-                }
+    private getCredentialContainers(): void {
+        AccountServiceClient.Instance.doGet("/credentials")
+            .then((response: AxiosResponse<CredentialContainer[]>) => {
+
+                const allContainers: CredContainerMap = new Map();
+                response.data.forEach(container => {
+                    let categoryMap = allContainers.get(container.category);
+                    if (!categoryMap) {
+                        categoryMap = new Map();
+                        allContainers.set(container.category, categoryMap);
+                    }
+                    categoryMap.set(container.type, container);
+                });
+
+                this.setState({credentialContainers: allContainers});
             });
     }
     
-    private handleTwoFactorSwitch = () => {
-        if (this.state.twoFactorEnabled) {
-            this.setState({twoFactorEnabled: false, twoFactorEnabledText: Msg.localize('twoFactorDisabled')})
-        } else {
-            this.setState({twoFactorEnabled: true, twoFactorEnabledText:  Msg.localize('twoFactorEnabled')})
-        }
-    }
-
-    private handleRemoveTOTP = () => {
-      AccountServiceClient.Instance.doDelete("/totp/remove")
+    private handleRemove = (credentialId: string, userLabel: string) => {
+      AccountServiceClient.Instance.doDelete("/credentials/" + credentialId)
         .then(() => {
-            this.setState({isTotpConfigured: false});
-            ContentAlert.success('successTotpRemovedMessage');
+            this.getCredentialContainers();
+            ContentAlert.success('successRemovedMessage', [userLabel]);
         });
     }
 
     public render(): React.ReactNode {
```

Comment:
Just a suggestion. The two-factor authentication looks like it's an independent credential type. If I understand it correctly, it works only with password. Couldn't we somehow visually hint the user that it's not independent? E.g. somehow place it in the same box as password?

Of course, for the tech preview it's ok as it is.

Reply:
> Just a suggestion. The two-factor authentication looks like it's an independent credential type. If I understand it correctly, it works only with password. Couldn't we somehow visually hint the user that it's not independent? E.g. somehow place it in the same box as password?
> 
> Of course, for the tech preview it's ok as it is.

@mposolda Do you have any feedback on this?